### PR TITLE
[SQL][MINOR][Refactor] Refactor on sql/core

### PR DESCRIPTION
**Pull request by da-liii**
## What changes were proposed in this pull request?
Only minor changes on Scala syntax.

## How was this patch tested?
Existing Tests

### CODE REVIEW

**Review by HyukjinKwon — COMMENTED** [review]
Looks mostly style changes. I wouldn't do this since it makes backporting harder. I think it's best to help review and leave some comments for such nits in the future.

**Comment by sadhen** [comment]
@HyukjinKwon  OK, most of changes are related to parens. I will consider to reset it back for backporting friendliness.

**Comment by HyukjinKwon** [comment]
I think you're mostly trying to target changes styles. If there is not actual benefit rather then just styles, I wouldn't do this and just help review other PRs.

File: sql/core/src/main/scala/org/apache/spark/sql/api/r/SQLUtils.scala [hunk 1]
```diff
@@ -188,7 +188,7 @@ private[sql] object SQLUtils extends Logging {
     dataType match {
       case 's' =>
         // Read StructType for DataFrame
-        val fields = SerDe.readList(dis, jvmObjectTracker = null).asInstanceOf[Array[Object]]
+        val fields = SerDe.readList(dis, jvmObjectTracker = null)
```
Comment (sadhen) [inline sql/core/src/main/scala/org/apache/spark/sql/api/r/SQLUtils.scala]: useless asInstanceOf

File: sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala [hunk 2]
```diff
@@ -95,7 +95,7 @@ object SQLMetrics {
 
   def createMetric(sc: SparkContext, name: String): SQLMetric = {
     val acc = new SQLMetric(SUM_METRIC)
-    acc.register(sc, name = Some(name), countFailedValues = false)
+    acc.register(sc, name = Some(name))
```
Comment (sadhen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala]: duplicated assignment

File: sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala [hunk 3]
```diff
@@ -147,7 +147,7 @@ private[sql] object SparkUserDefinedFunction {
       f: AnyRef,
       dataType: DataType,
       inputSchemas: Option[Seq[ScalaReflection.Schema]]): UserDefinedFunction = {
-    val udf = new UserDefinedFunction(f, dataType, inputSchemas.map(_.map(_.dataType)))
+    val udf = UserDefinedFunction(f, dataType, inputSchemas.map(_.map(_.dataType)))
```
Comment (sadhen) [inline sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala]: useless `new` for case class

**Comment by sadhen** [comment]
@HyukjinKwon  Sorry. Happened to find that some code is not elegant according to my taste. I have not got the fact that massive changes are not good for backporting. As a result, I spent some time trying the make the code better.

I have commented on the most important ones according to my standard. Please review. And the rest, I will revert.

**Review by HyukjinKwon — COMMENTED** [review]
It still looks mostly just style changes. Touching multiple files here and there brings backporting overhead in the future (say, this diff will causes some conflicts when we backport some patches into other branches that does not have this change). If the change is not worth enough like just style, I wouldn't go ahead.

There's one legitimate change that might cause a potential actual problem (I commented above), which I wouldn't propose a PR to fix one instance

**Comment by sadhen** [comment]
OK. Nevermind.

**Pull request closed without merging**